### PR TITLE
LTI endpoint for getting test review

### PR DESCRIPTION
**Pull request by Alroniks**
https://oat-sa.atlassian.net/browse/TAO-8490

Requires https://github.com/oat-sa/extension-tao-ltideliveryprovider/pull/202

### CODE REVIEW

**Comment by Alroniks** [comment]
Please, have a look at readme about LTI calls instructions.

File: controller/Review.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        if ($deliveryExecutionId !== null) {
+            $params['delivery_execution'] = $deliveryExecutionId;
+        }
+
+        $this->setClientRoute(_url('index', 'Review', 'taoReview', $params ?? []));
+        $this->composeView('delegated-view', null, 'pages/index.tpl', 'tao');
+    }
+
+}
```
Comment (SergiiTao) [inline controller/Review.php]: could you please add new line here and further down?

File: controller/Review.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        $launchData = LtiService::singleton()->getLtiSession()->getLaunchData();
+
+        /** @var LtiLaunchDataService $launchDataService */
+        $launchDataService = $this->getServiceLocator()->get(LtiLaunchDataService::SERVICE_ID);
+
+        /** @var LtiResultAliasStorage $ltiResultIdStorage */
+        $ltiResultIdStorage = $this->getServiceLocator()->get(LtiResultAliasStorage::SERVICE_ID);
+
+
+        $resultIdentifier = $launchData->hasVariable('lis_result_sourcedid')
```
Comment (SergiiTao) [inline controller/Review.php]: do we have a constant for `lis_result_sourcedid`?

File: controller/Review.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+namespace oat\taoReview\controller;
+
+use common_exception_Error;
+use oat\ltiDeliveryProvider\model\LtiLaunchDataService;
+use oat\ltiDeliveryProvider\model\LtiResultAliasStorage;
+use oat\taoLti\models\classes\LtiService;
+use tao_actions_SinglePageModule;
+
+/**
+ * Class Review
```
Comment (SergiiTao) [inline controller/Review.php]: i think this comment is quite useless :)

File: controller/Review.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+ * @package oat\taoReview\controller
+ */
+class Review extends tao_actions_SinglePageModule
+{
+    /**
+     * @throws \oat\taoLti\models\classes\LtiException
```
Comment (SergiiTao) [inline controller/Review.php]: lets get rid of full path namespaces/

File: controller/Review.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program; if not, write to the Free Software
+* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*
+* Copyright (c) 2016 (original work) Open Assessment Technologies SA;
```
Comment (Skyld) [inline controller/Review.php]: 2019

File: controller/Review.php (hunk 2 above)
Comment (Alroniks) [reply controller/Review.php]: Nope, unfortunately

File: controller/Review.php [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+ */
+class Review extends tao_actions_SinglePageModule
+{
+    /**
+     * @throws LtiException
+     * @throws LtiVariableMissingException
+     * @throws common_exception_Error
+     */
+    public function index()
+    {
```
Comment (SergiiTao) [inline controller/Review.php]: to much logick for a controller. it should be separated to the service

Comment (Skyld) [reply controller/Review.php]: Let's create a follow-up task for the next sprint for this

**Pull request merged**